src/content/doc-surrealql/statements/define/api.mdx

```diff
@@ -0,0 +1,170 @@
+---
+sidebar_position: 3
+sidebar_label: DEFINE API
+title: DEFINE API statement | SurrealQL
+description: A DEFINE API statement can be used to set endpoints with custom middleware and permissions.
+---
+import Since from '@components/shared/Since.astro'
+
+<Since v="v2.2.0" />
+
+# DEFINE API statement
+> [!CAUTION]
+> Currently, this is an experimental feature as such, it may be subject to breaking changes and may present unidentified security issues. Do not rely on this feature in production applications. To enable this, set the `SURREAL_CAPS_ALLOW_EXPERIMENTAL` [environment variable](/docs/surrealdb/cli/start/#experimental-capabilities) to `define_api`.
+
+The `DEFINE API` statements allows a custom endpoint to be created. Each endpoint created by a `DEFINE API` statement is located at the `/api/:namespace/:database/:endpoint_name` path. For example, an endpoint for the path `get_users` for the namespace `my_namespace` and database `my_database` will have the path `/api/my_namespace/my_database/get_users`.
+
+The response is an object with a combination of the following properties:
+* `raw` - A boolean that when set to `true` sends the response back as raw bytes or a string.
+* `status` - A valid HTTP status code.
+* `headers` - An object of valid header key value pairs.
+* `body` - A SurrealQL value, automatically encoded back into how the user requested it. When `raw` is set to `true`, this needs to be a string or bytes value.
+
+## Statement syntax
+
+```syntax title="SurrealQL Syntax"
+DEFINE API [ OVERWRITE | IF NOT EXISTS ] @endpoint
+    [ FOR @HTTP_method, .. ]
+    [ MIDDLEWARE @function, .. ]
+    [ THEN { @value } ]
+    [ PERMISSIONS [ NONE | FULL | @expression ]
+```
+
+
+`DEFINE API` is often used in conjunction with a [flag](/docs/surrealdb/security/capabilities/) or [environment variable](/docs/surrealdb/cli/env/) to disable arbitrary queries, thereby forcing record and anonymous users to interact with the database via API endpoints alone.
+
+## Example usage
+
+```surql
+DEFINE API "/test"
+    FOR get, post 
+        MIDDLEWARE
+            api::req::raw_body(false)
+        THEN {
+            RETURN {
+                status: 200,
+                body: {
+                    request: $request.body,
+                    response: "The server works"
+                },
+                headers: {
+                    'last-modified': time::now(),
+                    'expires': time::now() + 4d
+                }
+            };
+        };
+```
+
+An API endpoint can be tested using the `api::invoke` function, which takes either the path as a single string or the path along with a request body. It can also be tested via CURL or other means by directly [using the endpoint](/docs/surrealdb/integration/http/#ml-export) along with the namespace and database in the headers.
```

Comment:
```suggestion
An API endpoint can be tested using the [`api::invoke` function](/docs/surrealql/functions/database/api/#apiinvoke), which takes either the path as a single string or the path along with a request body. It can also be tested via CURL or other means by directly [using the endpoint](/docs/surrealdb/integration/http/#ml-export) along with the namespace and database in the headers.
```